docs/docsite/rst/dev_guide/developing_modules_general.rst

```diff
@@ -48,6 +48,133 @@ Creating a development environment (platform-agnostic steps)
    ``$ . venv/bin/activate && . hacking/env-setup``
 
 
+Creating an info module
+=======================
+
+Ansible gather information about the target machines using facts modules.
+If you find yourself trying to add ``state: info`` or ``state: list`` to an existing module that is often a sign that a new dedicated ``_facts`` or ``_info`` module is needed.
+
+Facts modules, are just like any other Ansible Module, with a few minor requirements:
+
+1. They MUST be named ``<something>_info`` or ``<something>_facts``, where <something> is singular
+2. They MUST return facts in the form of the :ref:`result dictionary<common_return_values>` so other modules can access them
+3. They MUST support :ref:`check_mode <check_mode_dry>`
+4. They MUST NOT make any changes to the system
+5. They MUST document the :ref:`return fields<return_block>` including :ref:`examples<examples_block>`
+
+To create an info module:
+
+1. Navigate to the correct directory for your new module: ``$ cd lib/ansible/modules/``
```

Comment:
This is only true for modules inside ansible/ansible. Since most people would be interested in creating info/facts modules in collections, I guess we should use `plugins/modules/` here?